File: src/main/java/gg/climb/lolobjects/InternalId.java
package gg.climb.lolobjects;

import org.immutables.value.Value;

/**
 * Representation of our internal database ids. You can obtain a new instance via
 * {@link ImmutableInternalId#of(int)}. However, this class should not be used in
 * most cases outside of the database.
 */
@SuppressWarnings("unused") // Actually used type parameter
@Value.Immutable(builder = false)
public abstract class InternalId<T> {
  @Value.Parameter
  public abstract int getId();
}

File: src/main/scala/gg/climb/lolobjects/InternalId.scala
package gg.climb.lolobjects

class InternalId[T](val id: String) {
  override def toString = s"InternalId=$id"
  override def equals(other: Any): Boolean = {
    other match {
      case internalId: InternalId[T] => this.id.equals(internalId.id)
      case _ => false
    }
  }
}
